From 09929db075f9eb8c5e8032529ef1115c2c9b6c96 Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <Nikolai.Kondrashov@redhat.com>
Date: Thu, 29 Jan 2015 14:08:12 +0200
Subject: [PATCH 1/1] Initialize child count in modcall_recurse

Initialize child count in modcall_recurse to explicitly handle the case of no
children, making the code at least slightly easier to read, if not actually
fixing an issue.

This fixes the following Coverity error:

Error: UNINIT (CWE-457):
freeradius-server-2.2.6/src/main/modcall.c:691: var_decl: Declaring variable "count" without initializer.
freeradius-server-2.2.6/src/main/modcall.c:727: uninit_use: Using uninitialized value "count".
---
 src/main/modcall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/modcall.c b/src/main/modcall.c
index 7e96562..d9cebe9 100644
--- a/src/main/modcall.c
+++ b/src/main/modcall.c
@@ -687,7 +687,7 @@ redo:
 
 	if ((c->type == MOD_LOAD_BALANCE) ||
 	    (c->type == MOD_REDUNDANT_LOAD_BALANCE)) {
-		int count;
+		int count = 0;
 		modcallable *this, *found;
 		modgroup *g;
 
-- 
2.1.4

